### PR TITLE
Add Modal widget

### PR DESCRIPTION
**Pull request by shkolin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #21

### CODE REVIEW

File: src/ModalCard.php [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        $html .= $this->renderBodyBegin() . "\n";
+
+        return $html;
+    }
+
+    protected function run(): string
+    {
+        $html = '';
+        $html .= $this->renderBodyEnd() . "\n";
+        $html .= $this->renderFooter() . "\n";
```
Comment (shkolin) [inline src/ModalCard.php:87]: @samdark

With this arrangement of the method `renderFooter ` adding options becomes impossible. Please tell me how to solve this issue?

```php
echo Modal::begin()
    ->footerOptions(['class' => 'bg-transparent'])
    ->start();
...
```

Comment (samdark) [reply src/ModalCard.php:87]: Doesn't the code above work well?

Comment (shkolin) [reply src/ModalCard.php:87]: > Doesn't the code above work well?

Yes, please take a look: https://github.com/yiisoft/yii-bulma/pull/20/files#diff-3a5fcb7d5f593e26c99e016ab2c8513bf9283d200e23ae60b9f6b11fde03e200R38

Comment (samdark) [reply src/ModalCard.php:93]: Seems it's a bug in base widget: https://github.com/yiisoft/widget/pull/21

**Comment by terabytesoftw** [comment]
@shkolin  Please change widget last update in: https://github.com/yiisoft/widget/pull/22, it should work now.

**Comment by shkolin** [comment]
> @shkolin Please change widget last update in: [yiisoft/widget#22](https://github.com/yiisoft/widget/pull/22), it should work now.
> 
> change:
> 
> ```
> $html = NavBar::widget()->begin();
> $html .= NavBar::end();
> ```

done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**